platform/ext/target/nordic_nrf/common/nrf5340/tfm_hal_its_encryption.c

```diff
@@ -0,0 +1,204 @@
+#include "platform/include/tfm_platform_system.h"
+#include "platform/include/tfm_hal_its.h"
+#include "psa/crypto.h"
+
+#include "nrf_cc3xx_platform_derived_key.h"
+#include "nrf_cc3xx_platform.h"
+/* TODO: Remove this after testing */
+#include "nrf_cc3xx_platform_kmu.h"
+#include <string.h>
+
+#define ITS_ENCRYPTION_SUCCESS 0
+
+#define HUK_KMU_SLOT 2
+#define HUK_KMU_SIZE_BITS 256
+
+static int32_t tfm_hal_its_aead_crypt(uint8_t *plaintext,
+                                      size_t plaintext_size,
```

Comment:
That is a lot of arguments, would it make sense to introduce a struct to hold some of this?
Or use tfm_hal_its_aead_ctx?

Reply:
Absolutely, I did that in the latest push.

Reply:
I could not find this change in the current files.